### DISCUSSION TITLE
FR: 求适配 Vertex AI

### DISCUSSION THREAD

**Discussion by fangyuan99** [CherryHQ/cherry-studio#1863] [Model & Provider | 模型和服务商] [answered]
求适配 Vertex AI 渠道，想用联网搜索的 gemini

**Comment by kangfenmao** [comment]
> 求适配 Vertex AI 渠道，想用联网搜索的 gemini

有没有文档可以帮助我参考一下

**Reply by jlong7398** [reply]
Vertex AI的稳定性更好，建议作者适配一下

**Comment by fangyuan99** [comment]
https://cloud.google.com/vertex-ai/docs

https://linux.do/t/topic/229642

**Comment by happyTonakai** [comment]
请问有进展吗，目前的gemini还是只能走ai studio

**Reply by fangyuan99** [reply]
开发者可能忘记了

**Comment by clionertr** [comment]
vertex的文档东一块西一块,模型数量还特别多,估计有的等了😭

**Comment by happyTonakai** [comment, marked as answer]
v1.4.5已支持

**Reply by Twilight-9** [reply]
> v1.4.5已支持

目前Vertex AI，Gemini的模型可用，Claude不可用，可能是因为Claude的设置不太一样？

<img width="519" alt="image" src="https://github.com/user-attachments/assets/54ede9c8-25c9-44c7-a30f-85642e972c24" />

<img width="1001" alt="image" src="https://github.com/user-attachments/assets/bd9d9d90-3851-4e73-ad61-acebbf81c71d" />

**Reply by DeJeune** [reply]
Claude要单独适配，耐心等待一下

**Reply by DeJeune** [reply]
> > v1.4.5已支持
> 
> 目前Vertex AI，Gemini的模型可用，Claude不可用，可能是因为Claude的设置不太一样？
> 
> <img alt="image" width="519" src="https://private-user-images.githubusercontent.com/62452364/458241039-54ede9c8-25c9-44c7-a30f-85642e972c24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA5MjM5MzAsIm5iZiI6MTc1MDkyMzYzMCwicGF0aCI6Ii82MjQ1MjM2NC80NTgyNDEwMzktNTRlZGU5YzgtMjVjOS00NGM3LWEzMGYtODU2NDJlOTcyYzI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjI2VDA3NDAzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVjYmZiNGZiNDRlMjRiOTc5ZTMxOGMxZDM3NWNhMjUxZTk3NGY0NjIzZDkzODFkNmU3YmM0NGUwNzk2ZDA4ZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nYfQV79mJuQNDXUlxHEyWP5GwOA_VeYF54JqH0OMwHk"> <img alt="image" width="1001" src="https://private-user-images.githubusercontent.com/62452364/458240788-bd9d9d90-3851-4e73-ad61-acebbf81c71d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA5MjM5MzAsIm5iZiI6MTc1MDkyMzYzMCwicGF0aCI6Ii82MjQ1MjM2NC80NTgyNDA3ODgtYmQ5ZDlkOTAtMzg1MS00ZTczLWFkNjEtYWNlYmJmODFjNzFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjI2VDA3NDAzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0ZTgwOWZiMjJlOWI3MWJiNmM2Y2I1MjBmOGZjYzkwNjE2NDY3NjIxYTJmZGNhNTIyOWU2OGZhYzQ0YTU5MGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RkEFf7MWMkHN8DuUAVJKIQ4XnZ68QvxD8CD1IS1Hq4Q">



> > v1.4.5已支持
> 
> 目前Vertex AI，Gemini的模型可用，Claude不可用，可能是因为Claude的设置不太一样？
> 
> <img alt="image" width="519" src="https://private-user-images.githubusercontent.com/62452364/458241039-54ede9c8-25c9-44c7-a30f-85642e972c24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA5MjM5MzAsIm5iZiI6MTc1MDkyMzYzMCwicGF0aCI6Ii82MjQ1MjM2NC80NTgyNDEwMzktNTRlZGU5YzgtMjVjOS00NGM3LWEzMGYtODU2NDJlOTcyYzI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjI2VDA3NDAzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVjYmZiNGZiNDRlMjRiOTc5ZTMxOGMxZDM3NWNhMjUxZTk3NGY0NjIzZDkzODFkNmU3YmM0NGUwNzk2ZDA4ZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nYfQV79mJuQNDXUlxHEyWP5GwOA_VeYF54JqH0OMwHk"> <img alt="image" width="1001" src="https://private-user-images.githubusercontent.com/62452364/458240788-bd9d9d90-3851-4e73-ad61-acebbf81c71d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA5MjM5MzAsIm5iZiI6MTc1MDkyMzYzMCwicGF0aCI6Ii82MjQ1MjM2NC80NTgyNDA3ODgtYmQ5ZDlkOTAtMzg1MS00ZTczLWFkNjEtYWNlYmJmODFjNzFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjI2VDA3NDAzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0ZTgwOWZiMjJlOWI3MWJiNmM2Y2I1MjBmOGZjYzkwNjE2NDY3NjIxYTJmZGNhNTIyOWU2OGZhYzQ0YTU5MGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RkEFf7MWMkHN8DuUAVJKIQ4XnZ68QvxD8CD1IS1Hq4Q">

https://github.com/CherryHQ/cherry-studio/pull/7564 看下这个pr好使不，我的账号调用Claude全部报错429

**Reply by fangyuan99** [reply]
可以支持开启关闭思考吗？
